app/models/project.rb

```diff
@@ -1,6 +1,12 @@
 class Project < KalibroClient::Entities::Processor::Project
   include KalibroRecord
+
   def self.latest(count = 1)
-    all.sort { |a,b| b.id <=> a.id }.first(count)
+    all.sort { |a,b| b.id <=> a.id }.select { |project| !project.attributes.hidden}.first(count)
+  end
+
+  def attributes
+    project_attributes = ProjectAttributes.find_by_project_id(self.id)
+    project_attributes.nil? ? ProjectAttributes.new : project_attributes
```

Comment:
project_attributes is not a class attribute.

Em qui, 26 de fev de 2015 às 13:43, Diego de Araújo Martinez Camarinha <
notifications@github.com> escreveu:

> In app/models/project.rb
> <https://github.com/mezuro/prezento/pull/183#discussion_r25442132>:
>
> >    def self.latest(count = 1)
> > -    all.sort { |a,b| b.id <=> a.id }.first(count)
> > +    all.sort { |a,b| b.id <=> a.id }.select { |project| !project.attributes.hidden}.first(count)
> > +  end
> > +
> > +  def attributes
> > +    project_attributes = ProjectAttributes.find_by_project_id(self.id)
> > +    project_attributes.nil? ? ProjectAttributes.new : project_attributes
>
> Why don't we make project_attributes ||=
> ProjectAttributes.find_by_project_id(self.id) ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mezuro/prezento/pull/183/files#r25442132>.
>